internal/cmdrender/executor.go

```diff
@@ -286,30 +286,99 @@ func (pn *pkgNode) runPipeline(ctx context.Context, hctx *hydrationContext, inpu
 		return input, nil
 	}
 
-	fnChain, err := fnChain(ctx, pl, pn.pkg.UniquePath)
+	mutatedResources, err := pn.runMutators(ctx, hctx, input)
+	if err != nil {
+		return nil, errors.E(op, pn.pkg.UniquePath, err)
+	}
+
+	if err = pn.runValidators(ctx, hctx, mutatedResources); err != nil {
+		return nil, errors.E(op, pn.pkg.UniquePath, err)
+	}
+	// print a new line after a pipeline running
+	pr.Printf("\n")
+	return mutatedResources, nil
+}
+
+// runMutators runs a set of mutators functions on given input resources.
+func (pn *pkgNode) runMutators(ctx context.Context, hctx *hydrationContext, input []*yaml.RNode) ([]*yaml.RNode, error) {
+	const op errors.Op = "pipeline.runMutators"
```

Comment:
For this `op`, since we don't want to expose internal implementation details to users, I suspect does it really have a use case? Is it only used in debug? Is there a debug mode for kpt? And why don't we directly add a real stack trace here?

Reply:
This is no longer useful, we took out the stacktrace printing. Taking it out.